sdk/core/azure-core-test/src/test/java/com/azure/core/test/TestProxyTests.java

```diff
@@ -414,16 +414,49 @@ public void testRecordWithRedirect() {
     }
 
     private RecordedTestProxyData readDataFromFile() {
-        String filePath = Paths.get(TestUtils.getRecordFolder().getPath(), this.testContextManager.getTestPlaybackRecordingName()) + ".json";
-
-        File recordFile = new File(filePath);
-        try (BufferedReader reader = Files.newBufferedReader(recordFile.toPath())) {
+        try {
+            File recordFile = new File(locateAssetJsonFilePath());
+            BufferedReader reader = Files.newBufferedReader(recordFile.toPath());
             return RECORD_MAPPER.readValue(reader, RecordedTestProxyData.class);
         } catch (IOException ex) {
             throw new UncheckedIOException(ex);
         }
     }
 
+    private String locateAssetJsonFilePath() throws IOException {
+        String commandLine = Paths.get(TestProxyDownloader.getProxyDirectory().toString(),
+            TestProxyUtils.getProxyProcessName()).toString();
+        Path engRepoRoot = TestUtils.getRepoRootResolveUntil(getTestClassPath(), "eng");
+        String targetRepoRoot = TestUtils.getRepoRootResolveUntil(getTestClassPath(), "target").toString();
+        ProcessBuilder builder = new ProcessBuilder(commandLine,
+            "config",
+            "locate",
+            "-a",
+            "sdk\\core\\azure-core-test\\assets.json",
+            "--storage-location",
+            engRepoRoot.toString());
```

Comment:
This may be the problem. This needs to be the root of the repo, not the `eng` folder.

Given that the proxy will be running out of storage location `<path-to-repo>/eng` it'll resolve to `<repo root>/eng/sdk/core/azure-core-test/assets.json` which won't resolve.